src/UI/Component/Trigger/TriggerAction.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace ILIAS\UI\Component\Trigger;
+use ILIAS\UI\Component\JavaScriptBindable;
+
+/**
+ * Interface TriggerAction
```

Comment:
You're right that the current implementation is far from optimal, you should not be able to modify an action object like this. I think that this could be solved by representing each action in its own class.